### ISSUE TITLE
Has Many not working as I expected. Bug or feature?

### ISSUE THREAD

**Issue by jtorvald** [gobuffalo/pop#150]
```
type Interests []Interest
type Interest struct {
	UserID     int64 `db:"user_id" json:"user_id"`
	InterestID uint8 `db:"interest_id" form:"interest_id" json:"interest_id"`
}
type User struct {
        Interests   Interests   `has_many:"user_interests" fk_id:"user_id"`
}
```
POP doesn't searches user_interests table, as documented:
> Books        Books     `has_many:"books" order_by:"title asc"`
> has_many: will load all records from the books table that have a column named user_id or the column specified with fk_id that matches the User.ID value.

Instead it looks in: "interests". Even if I make the user hold UserInterests UserInterests -> then still it INSERTs into table interests. So it looks like it uses the singular type and adds an S? What is "has_many" used for then?

**Comment by kteb** [comment]
In your type User do you have a field ID ? Is it a int64 (by default pop is creating UUID for ID, but you can change it of course)?

Making something more reproducible (maybe with the commands you used to generates those models or a link to a project where we can see your code could be useful). 

Having only part of your code make it really difficult to debug.

**Comment by jtorvald** [comment]
Hi! Thanks for the quick reply. Yes, it has an ID and it is an int64. But the problem is that it guesses the table from the **type Interest struct** +s and it does not use the has_many as table name. I now renamed that type to UserInterest and it works. The bad thing is that I can not reuse that type to reference from another entity. To store that in another table I would need to create another type.

Sorry but don't have an example at the moment. But is it correct that **has_many:** should be able to contain the table name where it needs to store/read it from?

**Comment by u007** [comment]
are you looking for many-to-many?
this associate involve third table
has_many only support for belongs to between child and parent without third table

**Comment by jtorvald** [comment]
@u007 no, I understand what has_many needs to do, but I assumed that the meta data would be used for table name and foreign key but that seems not the case. If I remember correctly, I saw ".id" hardcoded in the code.

**Comment by u007** [comment]
@jtorvald, if you mean support for primary_key option, yes this will be interesting option to supp Rt

**Comment by u007** [comment]
I will look into making primary_key option
